tests/unit/lms/views/helpers/frontend_app_test.py

```diff
@@ -35,89 +28,75 @@ def test_it_disables_grading_if_grading_is_disabled_for_assignment(
-        for propName in [
-            "userid",
-            "displayName",
-            "LISResultSourcedId",
-            "LISOutcomeServiceUrl",
-        ]:
-            assert propName in js_config["grading"]["students"][0]
-
-    def test_it_sets_course_and_assignment_names(self, grading_request):
-        js_config = {}
-        frontend_app.configure_grading(grading_request, js_config)
-
-        assert js_config["grading"]["courseName"] == "Test Course 101"
-        assert js_config["grading"]["assignmentName"] == "How to use Hypothesis"
-
-
-pytestmark = pytest.mark.usefixtures("grading_info_service")
-
-
-@pytest.fixture
-def grading_infos():
-    grading_infos = []
-
-    for index in range(3):
-        h_username = f"username_{index}"
-        h_display_name = f"Student {index}"
-        lis_result_sourcedid = f"lis_result_sourcedid_{index}"
-
-        grading_infos.append(
-            (
-                GradingInfo(
-                    h_username=h_username,
-                    h_display_name=h_display_name,
-                    lis_result_sourcedid=lis_result_sourcedid,
-                    lis_outcome_service_url="http://example.com/service_url",
-                ),
-                HUser("test_authority", h_username, h_display_name),
+        assert js_config["grading"] == {
+            "courseName": "Test Course 101",
+            "assignmentName": "How to use Hypothesis",
+            "students": [
+                {
+                    "LISOutcomeServiceUrl": "test_lis_outcomes_service_url_0",
+                    "LISResultSourcedId": "test_lis_result_sourcedid_0",
+                    "displayName": "test_h_display_name_0",
+                    "userid": "acct:test_h_username_0@TEST_AUTHORITY",
+                },
+                {
+                    "LISOutcomeServiceUrl": "test_lis_outcomes_service_url_1",
+                    "LISResultSourcedId": "test_lis_result_sourcedid_1",
+                    "displayName": "test_h_display_name_1",
+                    "userid": "acct:test_h_username_1@TEST_AUTHORITY",
+                },
+                {
+                    "LISOutcomeServiceUrl": "test_lis_outcomes_service_url_2",
+                    "LISResultSourcedId": "test_lis_result_sourcedid_2",
+                    "displayName": "test_h_display_name_2",
+                    "userid": "acct:test_h_username_2@TEST_AUTHORITY",
+                },
+            ],
+        }
+
+    @pytest.fixture
+    def grading_info_service(self, grading_info_service):
+        grading_info_service.get_by_assignment.return_value = [
+            mock.create_autospec(
+                GradingInfo,
+                instance=True,
+                spec_set=True,
+                lis_result_sourcedid=f"test_lis_result_sourcedid_{i}",
+                lis_outcome_service_url=f"test_lis_outcomes_service_url_{i}",
+                h_username=f"test_h_username_{i}",
+                h_display_name=f"test_h_display_name_{i}",
```

Comment:
One nice thing about using a mock here is you only need to set the `GradingInfo` attributes that `frontend_app.py` actually uses and not the others